### PR TITLE
#1666  and #1656

### PR DESCRIPTION
**Pull request by pazembrz**

### CODE REVIEW

File: backend/inspirehep/records/api/base.py [hunk 1]
```diff
@@ -363,6 +366,22 @@ def redirect_pids(self, pids):
                 self.redirect_pid(pid_type, pid_value)
             return pids
 
+    @classmethod
+    def delete_records_from_record_deleted(cls, data):
```
Comment (drjova) [inline backend/inspirehep/records/api/base.py:370]:
```suggestion
    def delete_records_from_ deleted_records(cls, data):
``` 
:)

File: backend/inspirehep/records/api/base.py [hunk 2]
```diff
@@ -387,6 +406,7 @@ def update(self, data, *args, **kwargs):
             if data.get("deleted"):
                 self.pidstore_handler.delete(self.id, self)
             else:
+                self.delete_records_from_record_deleted(data)
```
Comment (drjova) [inline backend/inspirehep/records/api/base.py:409]: what about create? 🤔

**Comment by drjova** [comment]
@pazembrz 👏

File: backend/inspirehep/records/api/base.py (hunk 1 above)
Comment (pazembrz) [reply backend/inspirehep/records/api/base.py:370]: :+1:

File: backend/inspirehep/records/api/base.py (hunk 2 above)
Comment (pazembrz) [reply backend/inspirehep/records/api/base.py:409]: Ah, you are right :+1:  . It's OK for literature (as it's calling update inside create) but not for other types.

File: backend/inspirehep/records/api/base.py [hunk 3]
```diff
@@ -363,6 +366,22 @@ def redirect_pids(self, pids):
...
+        # Hack for migrator in case new record takes pids from other records
+        # which should be deleted but they are not deleted yet.
+        for pid in data.get("deleted_records", []):
+            pid_type, pid_value = PidStoreBase.get_pid_from_record_uri(pid["$ref"])
+            try:
+                record_to_delete = cls.get_record_by_pid_value(
+                    pid_value, pid_type, original_record=True
+                )
+            except PIDDoesNotExistError:
+                # Pid is not registered!
```
Comment (drjova) [inline backend/inspirehep/records/api/base.py:380]: do we need this comment? :)

**Review by drjova — APPROVED** [review]
just the comment!

File: backend/inspirehep/records/api/base.py (hunk 3 above)
Comment (pazembrz) [reply backend/inspirehep/records/api/base.py]: Comment is not needed, but I added a log as this is potential error/bug so we want to have some kind of trace that this situation happened.

**Pull request merged**